crates/electrum/tests/test_electrum.rs

```diff
@@ -0,0 +1,240 @@
+use anyhow::Result;
+use bdk_chain::{
+    bitcoin::{hashes::Hash, Address, Amount, ScriptBuf, WScriptHash},
+    keychain::Balance,
+    local_chain::LocalChain,
+    ConfirmationTimeHeightAnchor, IndexedTxGraph, SpkTxOutIndex,
+};
+use bdk_electrum::{ElectrumExt, ElectrumUpdate};
+use electrsd::bitcoind::bitcoincore_rpc::RpcApi;
+use electrum_client::ElectrumApi;
+use std::time::Duration;
+use testenv::TestEnv;
+
+fn wait_for_block(env: &TestEnv, client: &electrum_client::Client) -> Result<()> {
+    client.block_headers_subscribe()?;
+    let mut delay = Duration::from_millis(64);
+
+    loop {
+        env.electrsd.trigger()?;
+        client.ping()?;
+        if client.block_headers_pop()?.is_some() {
+            return Ok(());
+        }
+
+        if delay.as_millis() < 512 {
+            delay = delay.mul_f32(2.0);
+        }
+        std::thread::sleep(delay);
+    }
+}
+
+fn get_balance(
+    recv_chain: &LocalChain,
+    recv_graph: &IndexedTxGraph<ConfirmationTimeHeightAnchor, SpkTxOutIndex<()>>,
+) -> Result<Balance> {
+    let chain_tip = recv_chain.tip().block_id();
+    let outpoints = recv_graph.index.outpoints().clone();
+    let balance = recv_graph
+        .graph()
+        .balance(recv_chain, chain_tip, outpoints, |_, _| true);
+    Ok(balance)
+}
+
+/// Ensure that [`ElectrumExt`] can sync properly.
+///
+/// 1. Mine 101 blocks.
+/// 2. Send a tx.
+/// 3. Mine extra block to confirm sent tx.
+/// 4. Check [`Balance`] to ensure tx is confirmed.
+#[test]
+fn scan_detects_confirmed_tx() -> Result<()> {
+    const SEND_AMOUNT: Amount = Amount::from_sat(10_000);
+
+    let env = TestEnv::new()?;
+    let client = electrum_client::Client::new(env.electrsd.electrum_url.as_str())?;
```

Comment:
Now #1306 is merged, I think we can just use `env.electrum_client()`.

Reply:
Getting the versions of the crates to match up right now is impossible. I think what you have is optimal for now.